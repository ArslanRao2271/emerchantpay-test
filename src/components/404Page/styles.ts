import { SxProps } from "@mui/material";
import { boxDisplay } from "../../appStyles";

const styles: Record<string, SxProps> = {
  pageNotFoundContainer: {
    ...boxDisplay("flex", "center", "center"),
    flexDirection: "column",
  },
};

export default styles;
