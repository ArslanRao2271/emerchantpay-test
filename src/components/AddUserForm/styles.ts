import { SxProps } from "@mui/material";
import { lightTheme } from "@theme/index";
import { boxDisplay } from "../../appStyles";

const { palette } = lightTheme;

export const formStyles: React.CSSProperties = {
  ...boxDisplay("flex", "center", "center"),
  flexDirection: "column",
  gap: "1rem",
  width: "90%",
};

const styles: Record<string, SxProps> = {
  formFieldsWrapper: {
    ...boxDisplay("flex", "center", "flex-start"),
    width: "100%",
    gap: "1rem",
    flexDirection: {
      lg: "row",
      md: "row",
      xs: "column",
    },
  },

  formTextField: {
    backgroundColor: palette.grey["50"],
    borderRadius: "0.25rem",

    fieldset: {
      border: `0.063rem solid ${palette.primary["200"]}`,
    },
    "&:hover .Mui-error fieldset": {
      borderColor: palette.brightRed.main,
    },
    "& .MuiOutlinedInput-root": {
      "&:hover fieldset": {
        border: `0.063rem solid ${palette.primary["200"]}`,
      },
      "&.Mui-focused fieldset": {
        borderColor: palette.primary["200"],
      },
      "&.Mui-error fieldset": {
        border: `0.063rem solid ${palette.brightRed.main}`,
      },
    },
    ".MuiFormHelperText-root": {
      margin: "0",
      paddingTop: "0.625rem",
      fontWeight: "700",
      fontSize: "0.875rem",
      lineHeight: "0.875rem",
    },
    svg: {
      fill: `${palette.gradient.main}`,
    },
    ".MuiOutlinedInput-input": {
      backgroundColor: palette.grey["50"],
      "::-ms-reveal": {
        display: "none",
      },
    },
  },
};

export default styles;
