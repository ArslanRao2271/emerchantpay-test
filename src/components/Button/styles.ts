import { SxProps } from "@mui/material";
import { lightTheme } from "@theme/index";

const { palette } = lightTheme;
const styles: Record<string, SxProps> = {
  common: {
    boxShadow: "none",
    textTransform: "capitalize",
    borderRadius: "0.25rem",
    letterSpacing: "0",
    ":disabled": {
      color: palette.grey["300"],
      svg: {
        fill: palette.grey["300"],
      },
    },
  },
  small: {
    height: "2rem",
    paddingY: "0.75rem",
    fontSize: "0.75rem",
    fontStyle: "normal",
    fontWeight: "400",
    lineHeight: "0.75rem",
    svg: {
      width: "0.75rem",
    },
  },
  medium: {
    height: "2.5rem",
    paddingX: "1.125rem",
    paddingY: "0.625rem",
    fontSize: "1rem",
    fontStyle: "normal",
    fontWeight: "800",
    lineHeight: "1.25rem",
    svg: {
      width: "1rem",
    },
  },
  large: {
    height: "3rem",
    paddingY: "0.875rem",
    paddingX: "2rem",
    fontSize: "1rem",
    fontStyle: "normal",
    fontWeight: "800",
    lineHeight: "1.25rem",
    svg: {
      width: "1rem",
    },
  },
  back: {
    border: "none",
    color: palette.grey["400"],
    svg: {
      fill: palette.grey["400"],
    },
    ":hover": {
      boxShadow: "none",
    },
  },
  primary: {
    backgroundColor: palette.secondary.main,
    color: palette.grey["500"],
    ":hover": {
      backgroundColor: {
        xs: palette.secondary.main,
        md: palette.secondary["300"],
      },
      boxShadow: "none",
      color: palette.grey["500"],
    },
    ":disabled": {
      backgroundColor: palette.grey["100"],
    },
  },
  secondary: {
    border: `0.0625rem solid ${palette.primary["200"]}`,
    backgroundColor: palette.primary["100"],
    color: palette.primary.main,
    boxShadow: "none",
    svg: {
      transition: "0.1s",
      fill: palette.primary.main,
      width: "1rem",
    },
    ":disabled": {
      backgroundColor: palette.grey["100"],
      border: "none",
    },
    ":hover": {
      backgroundColor: palette.primary.main,
      color: palette.grey["50"],
      boxShadow: "none",
      svg: {
        transition: "0.1s",
        fill: palette.grey["50"],
      },
    },
  },
  tertiary: {
    border: "none",
    backgroundColor: "transparent",
    color: palette.grey["500"],
    boxShadow: "none",
    ".MuiButton-startIcon": {
      margin: 0,
    },
    ":hover": {
      boxShadow: "none",
      border: "none",
      backgroundColor: "transparent",
    },
  },
  iconText: {
    border: `0.0625rem solid ${palette.primary["200"]}`,
    backgroundColor: palette.grey["50"],
    color: palette.primary.main,
    boxShadow: "none",
    svg: {
      transition: "0.1s",
      fill: palette.grey.main,
      width: "1rem",
    },
    ":hover": {
      boxShadow: "none",
      backgroundColor: palette.primary["100"],
      color: palette.primary.main,
      svg: {
        transition: "0.1s",
        fill: palette.primary.main,
      },
    },
  },
  primaryInverse: {
    backgroundColor: palette.grey["50"],
    border: `0.0625rem solid ${palette.grey["500"]}`,
    color: palette.grey["500"],
  },
  secondaryInverse: {
    backgroundColor: palette.grey["50"],
    border: `0.0625rem solid ${palette.primary["200"]}`,
    color: palette.primary.main,
  },
  tertiaryInverse: {
    backgroundColor: palette.grey["50"],
    border: "none",
    color: palette.secondary.main,
    ":hover": {
      boxShadow: "none",
      border: "none",
      backgroundColor: palette.grey["50"],
    },
  },
  iconTextInverse: {
    border: "none",
    backgroundColor: palette.grey["50"],
    color: palette.primary.main,
    boxShadow: "none",
    svg: {
      transition: "0.1s",
      fill: palette.primary.main,
      width: "1rem",
    },
    ":hover": {
      backgroundColor: palette.grey["50"],
    },
  },
};

export default styles;
