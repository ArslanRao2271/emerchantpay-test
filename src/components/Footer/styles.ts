import { SxProps } from "@mui/material";
import { lightTheme } from "@theme/index";

const { palette } = lightTheme;

const styles: Record<string, SxProps> = {
  labelStyle: {
    flexGrow: 1,
    textAlign: "center",
  },

  footerBar: {
    backgroundColor: palette.primary.main,
    position: "fixed",
    bottom: 0,
    top: "auto",
  },
};

export default styles;
