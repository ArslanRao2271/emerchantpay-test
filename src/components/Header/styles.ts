import { SxProps } from "@mui/material";
import { lightTheme } from "@theme/index";

const { palette } = lightTheme;

const styles: Record<string, SxProps> = {
  labelStyle: {
    flexGrow: 1,
  },

  appBar: {
    backgroundColor: palette.primary.main,
  },
};

export default styles;
