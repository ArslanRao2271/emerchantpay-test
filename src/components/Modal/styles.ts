import { SxProps } from "@mui/material";
import { lightTheme } from "@theme/index";

const { palette } = lightTheme;

const styles: Record<string, SxProps> = {
  modalBox: {
    position: "absolute",
    top: "50%",
    left: "50%",
    transform: "translate(-50%, -50%)",
    width: 400,
    bgcolor: palette.grey[50],
    border: "2px solid #000",
    p: 4,
  },
};

export default styles;
