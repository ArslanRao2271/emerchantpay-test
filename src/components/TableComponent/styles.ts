import { SxProps } from "@mui/material";
import { lightTheme } from "@theme/index";

const { palette } = lightTheme;

export interface ResponsiveProps {
  xl?: string;
  lg?: string;
  xs?: string;
  ss?: string;
  sm?: string;
  md?: string;
  mm?: string;
  ml?: string;
}

const boxDisplay = (
  display: string,
  justifyContent?: string,
  alignItems?: string
) => ({
  display,
  justifyContent,
  alignItems,
});

const textStyle = (
  fontWeight: number,
  fontSize: string | ResponsiveProps,
  lineHeight: string | ResponsiveProps
) => ({
  fontWeight,
  fontSize,
  lineHeight,
  whiteSpace: "pre-wrap",
});

const styles: Record<string, SxProps> = {
  headerCellText: {
    ...textStyle(400, "1rem", "1.25rem"),
    color: palette.grey[400],
  },

  rowTextStyle: {
    ...textStyle(400, "1rem", "1.25rem"),
    color: palette.grey.main ?? "",
  },

  viewDocumentCTA: {
    ...textStyle(800, { lg: "1rem", md: "1rem", xs: "0.875rem" }, "1.25rem"),
    color: palette.grey.main ?? "",
    borderLeft: `0.125rem solid ${palette.primary.main}`,
    paddingLeft: "0.5rem",
    cursor: "pointer",
    "& span": {
      display: {
        lg: "inline",
        md: "inline",
        xs: "none",
      },
    },
  },

  buttonsContainer: {
    ...boxDisplay("flex", "flex-start", "center"),
    gap: "0.25rem",
  },

  pastTenanciesDocumentLabelContainer: {
    ...boxDisplay("flex", "flex-start", "flex-start"),
    gap: "0.5rem",
  },

  docTitleStyle: {
    ...textStyle(400, { lg: "1rem", md: "1rem", xs: "0.75rem" }, "0.875rem"),
    color: palette.grey.main ?? "",
  },

  docSubTitleStyle: {
    ...textStyle(300, "0.75rem", "0.875rem"),
    color: palette.grey[600],
    mt: "0.3rem",
  },
};

export default styles;
