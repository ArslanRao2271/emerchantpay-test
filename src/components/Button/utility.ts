import React from "react";
import { lightTheme } from "@theme/index";
// import ArrowBackIcon from "@mui/icons-material/ArrowBack";
// import HorizontalRuleIcon from "@mui/icons-material/HorizontalRule";
// import ArrowForwardIcon from "@mui/icons-material/ArrowForward";
import styles from "./styles";

const {
  back,
  primary,
  secondary,
  tertiary,
  primaryInverse,
  secondaryInverse,
  tertiaryInverse,
  iconTextInverse,
} = styles;

const { palette } = lightTheme;

export const iconColor = (color?: string) => {
  if (color && palette[color]) {
    return palette[color].main;
  }

  return palette.primary.main;
};

export const getBtnStyles = (
  variant: string,
  color: string,
  theme?: string,
  startIcon?: React.ReactNode
) => {
  if (variant === "back") {
    return back;
  }

  if (color === "secondary" || theme === "dark") {
    if (startIcon !== undefined) {
      return iconTextInverse;
    }

    if (variant === "primary") {
      return primaryInverse;
    }

    if (variant === "secondary") {
      return secondaryInverse;
    }

    if (variant === "tertiary") {
      return tertiaryInverse;
    }

    return {};
  }

  if (startIcon !== undefined) {
    return secondary;
  }

  if (variant === "primary" && color === "primary") {
    return primary;
  }

  if (variant === "secondary" && color === "primary") {
    return secondary;
  }

  if (variant === "tertiary" && color === "primary") {
    return tertiary;
  }

  return {};
};

export const getIcon = (
  variant: string,
  isHover: boolean,
  color: string,
  StartIcon: React.ReactNode
) => {
  if (StartIcon !== undefined) {
    return StartIcon;
  }

  if (variant === "tertiary") {
    if (isHover) {
      return StartIcon;
    }

    return StartIcon;
  }

  if (variant === "back") {
    return StartIcon;
  }

  return undefined;
};
