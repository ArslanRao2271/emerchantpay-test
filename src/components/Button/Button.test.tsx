import { render, screen, fireEvent } from "@testing-library/react";
import AddIcon from "@mui/icons-material/Add";
import { lightTheme } from "@theme/index";
import ArrowBackIcon from "@mui/icons-material/ArrowBack";
import HorizontalRuleIcon from "@mui/icons-material/HorizontalRule";
import ArrowForwardIcon from "@mui/icons-material/ArrowForward";
import styles from "./styles";
import { getBtnStyles, getIcon, iconColor } from "./utility";
import { Button } from ".";

const { palette } = lightTheme;

const colorTest = [
  { color: "", expectation: palette.primary.main },
  { color: "secondary", expectation: palette.secondary.main },
  { color: "red", expectation: palette.primary.main },
  { color: "grey", expectation: palette.grey.main },
];

const VARIANTS = [
  { variant: "primary", color: "primary", expectation: styles.primary },
  { variant: "secondary", color: "primary", expectation: styles.secondary },
  { variant: "tertiary", color: "primary", expectation: styles.tertiary },
  { variant: "tertiary", color: "primary", expectation: styles.tertiary },
  { variant: "text", color: "secondary", expectation: {} },
  { variant: "text", color: "primary", expectation: {} },
  {
    variant: "",
    color: "",
    startIcon: <AddIcon />,
    expectation: styles.secondary,
  },
  {
    variant: "",
    color: "secondary",
    startIcon: <AddIcon />,
    expectation: styles.iconTextInverse,
  },
  {
    variant: "primary",
    color: "secondary",
    expectation: styles.primaryInverse,
  },
  {
    variant: "secondary",
    color: "secondary",
    expectation: styles.secondaryInverse,
  },
  {
    variant: "tertiary",
    color: "secondary",
    expectation: styles.tertiaryInverse,
  },
  { variant: "back", color: "secondary", expectation: styles.back },
];

const START_ICON = [
  {
    startIcon: <HorizontalRuleIcon />,
    color: "primary",
    hover: false,
    variant: "tertiary",
    expect: <HorizontalRuleIcon />,
  },
  {
    startIcon: <ArrowForwardIcon />,
    color: "primary",
    hover: true,
    variant: "tertiary",
    expect: <ArrowForwardIcon />,
  },
  {
    startIcon: <AddIcon />,
    color: "primary",
    hover: false,
    variant: "tertiary",
    expect: <AddIcon />,
  },
  {
    startIcon: <ArrowBackIcon />,
    color: "primary",
    hover: false,
    variant: "back",
    expect: <ArrowBackIcon />,
  },
  {
    startIcon: undefined,
    color: "primary",
    hover: false,
    variant: "text",
    expect: undefined,
  },
];

describe("Button test cases", () => {
  test("test if Button is in document", () => {
    render(
      <Button variant="primary" color="primary" theme="light">
        Hello
      </Button>
    );
    expect(screen.getByRole("button")).toBeInTheDocument();
  });

  test("test if label works", () => {
    const TEXT = "Hello";

    render(<Button variant="primary">{TEXT}</Button>);

    expect(screen.queryAllByText(TEXT).length).toBeTruthy();
  });

  test("variant as children if children not defined", () => {
    const VARIANT_TEXT = "secondary";

    render(<Button variant={VARIANT_TEXT} />);

    expect(screen.queryAllByText(VARIANT_TEXT).length).toBeTruthy();
  });

  test("test if svg is in document if variant=tertiary", () => {
    render(<Button variant="tertiary">hello</Button>);

    const button = screen.getByText("hello");

    fireEvent.mouseOver(button);
    fireEvent.mouseLeave(button);
    expect(button).toContainHTML("</svg>");
  });

  test("test if svg is in document if variant=tertiary", () => {
    render(<Button variant="tertiary">hello</Button>);

    const button = screen.getByText("hello");

    fireEvent.mouseOver(button);
    fireEvent.mouseLeave(button);
    expect(button).toContainHTML("</svg>");
  });

  colorTest.forEach((testCase) => {
    test(`if color = ${testCase.color} given, then return ${testCase.expectation}`, () => {
      const result = iconColor(testCase.color);
      const expected = testCase.expectation;

      expect(result).toEqual(expected);
    });
  });

  VARIANTS.forEach((testCase) => {
    test(`if variant = ${testCase.variant} & color = ${
      testCase.color
    } & startIcon = ${testCase.startIcon} given, then return ${JSON.stringify(
      testCase.expectation
    )}`, () => {
      const result = getBtnStyles(
        testCase.variant,
        testCase.color,
        "light",
        testCase.startIcon
      );
      const expected = testCase.expectation;

      expect(result).toEqual(expected);
    });
  });

  START_ICON.forEach((testCase) => {
    test(`if getIcon with ${JSON.stringify(testCase)} `, () => {
      const result = getIcon(
        testCase.variant,
        testCase.hover,
        testCase.color,
        testCase.startIcon
      );

      expect(result).toEqual(testCase.expect);
    });
  });
});
