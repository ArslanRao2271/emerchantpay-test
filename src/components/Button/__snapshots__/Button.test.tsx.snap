// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button test cases Should match previous snapshot 1`] = `
<DocumentFragment>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-primary MuiButton-primaryPrimary MuiButton-sizeMedium MuiButton-primarySizeMedium css-n76phm-MuiButtonBase-root-MuiButton-root"
    tabindex="0"
    type="button"
  >
    Hello
    <span
      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
    />
  </button>
</DocumentFragment>
`;
